app/models/commodity.rb

```diff
@@ -0,0 +1,29 @@
+class Commodity
+  attr_reader :code
+
+  def initialize(code:, service: :uk)
+    @code = code
+    @service = service
+  end
+
+  def description
+    fetch.data.attributes.description
+  end
+
+  def fetch
+    @fetch ||= begin
+                 commodity = Uktt::Commodity.new(options)
+                 commodity.retrieve
+                 commodity.response
+               end
```

Comment:
Because this method actually returns a response, I would suggest naming this method `response` instead. I think it reads better.

Reply:
Yeah, there was something fishy about it to me, too. I wanted it to have a verb as a name rather than a noun to reflect that it was doing something. But I think it's fine for it to feel like a getter.